pipeline-steps/templates/jobs/fortify-scan-job.yaml

```diff
@@ -15,46 +15,58 @@ parameters:
-          fortifyUsername: $(fortifyUsername)
-          fortifyPassword: $(fortifyPassword)
-          fortifyBaseUrl: ${{ parameters.fortifyBaseUrl }}
-          applicationName: ${{ parameters.applicationName }}
-          rootFolder: ${{ parameters.rootFolder }}
-          qualityGateLevel: ${{ parameters.qualityGateLevel }}
-          technologyStack: ${{ parameters.technologyStack }}
-          technologyVersion: ${{ parameters.technologyVersion }}
-          retryLimit: ${{ parameters.retryLimit }}
+  - ${{ if or(eq(variables['Build.SourceBranch'], 'refs/heads/master'), eq(variables['Build.SourceBranch'], 'refs/heads/main'), startsWith(variables['Build.SourceBranch'], 'refs/pull/')) }}:
+    - job: PrepareFortifySecrets
+      dependsOn: ${{ parameters.dependsOn }}
+      pool: ${{ parameters.vaultAgentPool }}
+      steps:
+        - checkout: none
+        - task: AzureCLI@2
+          name: GetFortifyCredentials
+          displayName: "Get Fortify Credentials"
+          inputs:
+            azureSubscription: ${{ parameters.azureSubscriptionEndpoint }}
+            scriptType: "bash"
+            scriptLocation: "inlineScript"
+            inlineScript: |
+              fortifyUsername=$(az keyvault secret show --vault-name "${{ parameters.azureVault }}" --name "${{ parameters.fortifyUsernameSecret }}" --query "value" -o tsv)
+              fortifyPassword=$(az keyvault secret show --vault-name "${{ parameters.azureVault }}" --name "${{ parameters.fortifyPasswordSecret }}" --query "value" -o tsv)
+              echo "##vso[task.setvariable variable=FortifyUsername;issecret=true;isOutput=true;]$fortifyUsername"
+              echo "##vso[task.setvariable variable=FortifyPassword;issecret=true;isOutput=true;]$fortifyPassword"
+    - job: RunFortifyScan
+      dependsOn: ['PrepareFortifySecrets']
+      variables:
+        fortifyUsername: $[ dependencies.PrepareFortifySecrets.outputs['GetFortifyCredentials.FortifyUsername'] ]
+        fortifyPassword: $[ dependencies.PrepareFortifySecrets.outputs['GetFortifyCredentials.FortifyPassword'] ]
+      ${{ if eq(parameters.scanAgentPool, '') }}:
+        pool:
+          vmImage: ubuntu-latest
+      ${{ if ne(parameters.scanAgentPool, '') }}:
+        pool: ${{ parameters.scanAgentPool }}
+      steps:
+        - checkout: self
+          clean: true
+        - template: ../steps/fortify-scan.yaml
+          parameters:
+            fortifyUsername: $(fortifyUsername)
+            fortifyPassword: $(fortifyPassword)
+            fortifyBaseUrl: ${{ parameters.fortifyBaseUrl }}
+            applicationName: ${{ parameters.applicationName }}
+            rootFolder: ${{ parameters.rootFolder }}
+            qualityGateLevel: ${{ parameters.qualityGateLevel }}
+            technologyStack: ${{ parameters.technologyStack }}
+            technologyVersion: ${{ parameters.technologyVersion }}
+            retryLimit: ${{ parameters.retryLimit }}
+  - ${{ else }}:
+    - job: RunFortifyScan
```

Comment:
Needed to avoid breaking change for existing pipelines that depend on RunFortifyScan. Now if the scan is not needed, this job will return success.